prover/crates/lib/prover_dal/src/fri_prover_dal.rs

```diff
@@ -429,27 +431,47 @@ impl FriProverDal<'_, '_> {
             )
             .fetch_all(self.storage.conn())
             .await
-            .unwrap();
-
-            let mut result = HashMap::new();
-
-            for row in &rows {
-                let stats: &mut JobCountStatistics = result
-                    .entry(JobIdentifiers {
-                        circuit_id: row.circuit_id as u8,
-                        aggregation_round: row.aggregation_round as u8,
-                        protocol_version: row.protocol_version as u16,
-                        protocol_version_patch: row.protocol_version_patch as u32,
-                    })
-                    .or_default();
+            .unwrap()
+            .iter()
+            .map(|row| {
+                let mut queued = 0;
+                let mut in_progress = 0;
                 match row.status.as_ref() {
-                    "queued" => stats.queued = row.count as usize,
-                    "in_progress" => stats.in_progress = row.count as usize,
-                    _ => (),
-                }
-            }
+                    "queued" => queued = row.count as usize,
+                    "in_progress" => in_progress = row.count as usize,
+                    _ => unreachable!("query filtering for 'queued' and 'in_progress' only"),
+                };
+                let job_count_statistics = JobCountStatistics {
+                    queued,
+                    in_progress,
+                };
+                let protocol_semantic_version = ProtocolSemanticVersion::new(
+                    ProtocolVersionId::try_from(row.protocol_version as u16)
+                        .expect("DB protocol version is broken"),
+                    VersionPatch(row.protocol_version_patch as u32),
+                );
 
-            result
+                // BEWARE, HERE BE DRAGONS.
+                // In database, the `circuit_id` stored is the circuit for which the aggregation is done,
+                // not the circuit which is running.
+                // There is a single node level aggregation circuit, which is circuit 2.
+                // This can aggregate multiple leaf nodes (which may belong to different circuits).
+                // This "conversion" is a forced hacky  way to use `circuit_id` 2 for nodes.
+                // A proper fix will be later provided to solve this once new auto-scaler is in place.
+                let circuit_id = if row.aggregation_round == 2 {
+                    2
+                } else {
+                    row.circuit_id as u8
+                };
```

Comment:
A bit safer way to do it would be to define the conversion of `row` (or its data) to the `CircuitProverStatsEntry`. This conversion would be a pure function and can be covered with a couple of tests for this specific behavior.